bots/sg_rules_delete_by_scope.py

```diff
@@ -0,0 +1,124 @@
+"""
+## sg_rules_delete_by_scope
+What it does: Deletes all rules on a security group with a specific scope, port and protocol are optional
+Usage: AUTO: sg_rules_delete_by_scope <scope> <direction> <port|*> <protocol|*>
+
+Example: AUTO: sg_rules_delete_by_scope 0.0.0.0/0 inbound 22 tcp
+Parameters:
+    scope: a.b.c.d/e
+    direction: inbound/ outbound
+    port: number/ *
+    protocol: TCP/ UDP/ *
+-When '*' is any value of the parameter
+
+Other Examples:
+    all rules with 1.0.0.0/16 scope will be deleted for any port and protocol:
+    sg_rules_delete_by_scope 1.0.0.0/16 inbound * *
+
+    all rules with 0.0.0.0/0 scope will be deleted for port 22 and any protocol:
+    sg_rules_delete_by_scope 0.0.0.0/0 inbound 22 *
+Notes :
+    -  before running this bot, ensure that your applications will work correctly without those rules
+    - if a port is in a port range the rule wont be deleted ! use * on port parameter to delete the rule for any port
+Limitations: IPv6 is not supported
+
+"""
+
+import re
+
+PORT_TO = 'portTo'
+PORT_FROM = 'port'
+PROTOCOL = 'protocol'
+SCOPE = 'scope'
+ALL_TRAFFIC_PORT = 0
+ALL_TRAFFIC_PROTOCOL = '-1'
+
+"""
+checks for ip validity , fix it to be right
+"""
+
+
+def fix_malformed(rule):
+    ip = re.split('/|\.', str(rule[SCOPE]))  # break ip to blocks
+    rule[SCOPE] = ip[0] + '.' + ip[1] + '.' + ip[2] + '.' + ip[3] + '/' + ip[4]
+    pass
+
+
+"""
+creates & removes the specified rules from a security group 
+"""
+
+
+def delete_sg(sg, sg_id, rule,direction, text_output):
+    # make sure that scope is in CIDR notation for example, 203.0.113.0/24
+    fix_malformed(rule)
```

Comment:
give more informative name, something like verify_scope_is_cidr

Reply:
name changed to verify_scope_is_cidr